pkg/target/target.go

```diff
@@ -426,6 +430,65 @@ func convertToLabelSelector(object map[string]interface{}) (*metav1.LabelSelecto
 	return obj, nil
 }
 
+// TODO: can we use generic for Unmarshal after go 1.18?
+func convertToMatch(object map[string]interface{}) (*match.Match, error) {
+	j, err := json.Marshal(object)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not convert unknown object to JSON")
+	}
+	obj := &match.Match{}
+	if err := json.Unmarshal(j, obj); err != nil {
+		return nil, errors.Wrap(err, "Could not convert JSON to Match")
+	}
+	return obj, nil
+}
+
+// ToMatcher converts .spec.match in mutators to Matcher.
 func (h *K8sValidationTarget) ToMatcher(u *unstructured.Unstructured) (constraints.Matcher, error) {
-	panic("not implemented")
+	obj, found, err := unstructured.NestedMap(u.Object, "spec", "match")
+	if err != nil {
+		return nil, err
+	}
+	if found && obj != nil {
+		match, err := convertToMatch(obj)
+		if err != nil {
+			return nil, err
+		}
+		return &Matcher{match}, nil
+	}
+	return nil, fmt.Errorf("%s %s has no match found", u.GetKind(), u.GetName())
 }
+
+type Matcher struct {
```

Comment:
Add TODO for adding backlink to the gatekeeper target cache. (This will be done in a different PR)